base_layer/core/src/transactions/transaction_components/encrypted_data.rs

```diff
@@ -84,25 +78,38 @@ impl EncryptedData {
         value: MicroTari,
         mask: &PrivateKey,
     ) -> Result<EncryptedData, EncryptedDataError> {
-        let mut openings = value.as_u64().to_le_bytes().to_vec();
-        openings.append(&mut mask.to_vec());
-        let aead_payload = Payload {
-            msg: openings.as_slice(),
-            aad: Self::TAG,
-        };
+        // Encode the value and mask
+        let mut bytes = Zeroizing::new([0u8; size_of::<u64>() + PrivateKey::KEY_LEN]);
+        bytes[..size_of::<u64>()].clone_from_slice(value.as_u64().to_le_bytes().as_ref());
+        bytes[size_of::<u64>()..].clone_from_slice(mask.as_bytes());
+
+        // Set up a buffer that also has room for the AEAD tag
+        let mut buffer = HeaplessVec::<u8, { size_of::<u64>() + PrivateKey::KEY_LEN + size_of::<Tag>() }>::new();
+        buffer.extend_from_slice(bytes.as_slice()).map_err(|_| EncryptedDataError::BufferError)?;
```

Comment:
Should be able to copy the data straight into `buffer`
```suggestion
        // Set up a buffer containing the value, mask and with enough room for the AEAD tag
        let mut buffer = HeaplessVec::<u8, { size_of::<u64>() + PrivateKey::KEY_LEN + size_of::<Tag>() }>::new();
        buffer.extend_from_slice(value.as_u64().to_le_bytes().as_ref()).map_err(|_| EncryptedDataError::BufferError)?;
        buffer.extend_from_slice(mask.as_bytes()).map_err(|_| EncryptedDataError::BufferError)?;;
        
```

Reply:
Good catch! Will update.

Reply:
Working on a change that uses an array instead, which should allow for `Zeroizing` wrappers.